pinot-plugins/pinot-metrics/pinot-dropwizard/src/main/java/org/apache/pinot/plugin/metrics/dropwizard/DropwizardMetricName.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.plugin.metrics.dropwizard;
+
+import org.apache.pinot.spi.metrics.PinotMetricName;
+
+
+public class DropwizardMetricName implements PinotMetricName {
+  private final Class _klass;
+  private final String _metricName;
+
+  public DropwizardMetricName(Class<?> klass, String name) {
+    _klass = klass;
+    _metricName = name;
+  }
+
+  public DropwizardMetricName(String metricName) {
+    this(DropwizardMetricName.class, metricName);
+  }
+
+  @Override
+  public String getMetricName() {
+    return _metricName;
+  }
+
+  /**
+   * Overrides equals method by calling the equals from the actual metric name.
+   */
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null || !(obj instanceof DropwizardMetricName)) {
```

Comment:
trick: instanceof actually also checks null. If obj is null, it's false. so just need
```
if (!(obj instanceof DropwizardMetricName))
```